### PR TITLE
Changed WP desc, stable tag, direct file access, use nonce, enqueue scripts, etc.

### LINKED ISSUE #1: Test alpha forms
Many of our form types, such as jest do not have proper testing. Provide a documented use of this for HCaptcha developer credit.

### LINKED ISSUE #2: use wp_enqueue commands: plugin is using <style> and/or <link> tags to insert CSS/JS
Should be using the built in functions for this:

https://developer.wordpress.org/reference/functions/wp_enqueue_script/
https://developer.wordpress.org/reference/functions/wp_enqueue_style/

On the admin pages it'll want to use the admin enqueues

https://developer.wordpress.org/reference/hooks/admin_enqueue_scripts/
https://developer.wordpress.org/reference/hooks/admin_print_scripts/
https://developer.wordpress.org/reference/hooks/admin_print_styles/

Example:

function hcap_captcha_script

### LINKED ISSUE #3: sanitize, escape, and validate POST calls
Examples:

$hcap_lf_status = $_POST[$hcap_lf_status_n];
update_option($hcap_lf_status_n, $hcap_lf_status);

When you include POST/GET/REQUEST/FILE calls in your plugin, it's important to sanitize, validate, and escape them. The goal here is to prevent a user from accidentally sending trash data through the system, as well as protecting them from potential security issues.

SANITIZE: Data that is input (either by a user or automatically) must be sanitized. This lessens the possibility of XSS vulnerabilities and MITM attacks where posted data is subverted.

VALIDATE: All data should be validated as much as possible. Even when you sanitize, remember that you don’t want someone putting in ‘dog’ when the only valid values are numbers.

ESCAPE: Data that is output must be escaped properly, so it can't hijack admin screens. There are many esc_*() functions you can use to make sure you don't show people the wrong data.

To help you with this, WordPress comes with a number of sanitization and escaping functions. You can read about those here:

https://developer.wordpress.org/plugins/security/securing-input/
https://developer.wordpress.org/plugins/security/securing-output/
Remember: You must use the MOST appropriate functions for the context. If you’re sanitizing email, use sanitize_email(), if you’re outputting HTML, use esc_html(), and so on.

Clean everything, check everything, escape everything, and never trust the users to always have input sane data.

### LINKED ISSUE #4: use Nonces and/or check permissions on POST calls
Please add a nonce to POST calls to prevent unauthorized access.

Keep in mind, check_admin_referer alone is NOT bulletproof security. Do not rely on nonces for authorization purposes. Use current_user_can() in order to prevent users without the right permissions from accessing things.

If you use wp_ajax, remember they ALSO need a nonce check.

The following links may assist you in develoment:

https://developer.wordpress.org/plugins/security/nonces/
https://developer.wordpress.org/plugins/javascript/ajax/#nonce

### LINKED ISSUE #5: ensure we disable Direct File Access for all plugin files
You can avoid this by putting this code at the top of all php files:

if ( ! defined( 'ABSPATH' ) ) exit; // Exit if accessed directly

For files that only contain a PHP class the risk of something funky happening when directly accessed is pretty small. For files that contain procedural code, functions and function calls, the chance of security risks is a lot bigger.

### LINKED ISSUE #6: Update "Stable tag" value in the readme.txt
The "Stable tag" value in the readme.txt should be the tag we'll create in the SVN repo and should be related to the plugin version in the plugin PHP file header.

(Credit: Felipe Elia)

### LINKED ISSUE #7: "Wordpress" in the plugin description should have a capital P.
Minor style point.

(Credit: Felipe Elia)

### LINKED ISSUE #8: Clean up code indentation, maybe autolint everything
For example, in the first lines of: https://github.com/hCaptcha/hcaptcha-wordpress-plugin/blob/master/bbp/bbp-reply.php

(Credit: Felipe Elia)

### PR DESCRIPTION
**Pull request by phpwebdev11**
Closes #8 
Closes #7 
Closes #6 
Closes #5 
Closes #4
Closes #3 
Closes  #2 
Closes #1

### CODE REVIEW

File: hcaptcha.php [hunk 1]
```diff
@@ -75,11 +73,12 @@
 // add the hcaptcha script to footer
 function hcap_captcha_script(){
 	global $hcap_language;
-    $output = '<script src="https://hcaptcha.com/1/api.js?hl='.$hcap_language.'" async defer></script>';
-    echo $output;
+    
+    wp_enqueue_script( 'hcpatcha-script', '//hcaptcha.com/1/api.js?hl='.$hcap_language, array(), false, true );
```
Comment (e271828-) [inline hcaptcha.php]: Typo: hcpatcha-script -> hcaptcha-script

File: cf7/hcaptcha-cf7.php [hunk 2]
```diff
@@ -33,12 +29,11 @@ function enqueue_hcap_cf7_script() {
                         grecaptcha.reset(widgetIds[i]);
                     }
                 });
-            })(jQuery);
-        </script>
-        <script src="<?php echo $hcap_cf7_script_url; ?>" async defer></script>
-        <?php
+            })(jQuery);";
+
+        wp_add_inline_script( 'hcpatcha-script', $script );
```
Comment (e271828-) [inline cf7/hcaptcha-cf7.php]: Typo: hcpatcha-script -> hcaptcha-script

File: nf/templates/fields-hcaptcha.html [hunk 3]
```diff
@@ -1,5 +1,7 @@
 <script id="tmpl-nf-field-hcaptcha" type="text/template">
 	<input id="nf-field-{{{ data.id }}}" name="nf-field-{{{ data.id }}}" class="{{{ data.renderClasses() }}} nf-element" type="hidden" value="{{{ data.value }}}" />
 
-	<div class="h-captcha" data-sitekey="{{{ data.hcaptcha_key }}}" data-theme="{{{ data.hcaptcha_theme }}}" data-size="{{{ data.hcaptcha_size }}}" data-fieldid="{{{ data.id }}}"></div>
-</script>
\ No newline at end of file
+	<div id="nf-hcpatcha" class="h-captcha" data-sitekey="{{{ data.hcaptcha_key }}}" data-theme="{{{ data.hcaptcha_theme }}}" data-size="{{{ data.hcaptcha_size }}}" data-fieldid="{{{ data.id }}}"></div>
```
Comment (e271828-) [inline nf/templates/fields-hcaptcha.html]: Typo: hcpatcha-> hcaptcha

File: nf/ninja-forms-hcaptcha.php [hunk 4]
```diff
@@ -16,7 +20,14 @@ function hcap_nf_template_file_paths( $paths ){
...
 }, 10, 1);
-wp_enqueue_script( 'nf-hcaptcha-js', plugin_dir_url( __FILE__ ) . 'nf-hcaptcha.js', array('nf-front-end' ), '1.0.0' );  
\ No newline at end of file
+
+add_action( 'wp_enqueue_scripts', 'hcap_nf_captcha_script' );
+
+function hcap_nf_captcha_script(){
+	wp_enqueue_script( 'nf-hcaptcha-js', plugin_dir_url( __FILE__ ) . 'nf-hcaptcha.js', array('nf-front-end' ), '1.0.0', true );  
+	wp_add_inline_script( 'nf-hcaptcha-js', 'setTimeout(function(){window.grecaptcha.render("nf-hcpatcha")}, 1000);' );
+}
```
Comment (posix4e) [inline nf/ninja-forms-hcaptcha.php]: another hcpatcha

File: readme.txt [hunk 5]
```diff
@@ -4,11 +4,11 @@ Tags: captcha, hcaptcha, make money with captcha, recaptcha, human captcha
 Requires at least: 4.0.1  
 Tested up to: 5.1.1  
 Requires PHP: 5.2.4  
-Stable tag: 4.9  
+Stable tag: trunk
```
Comment (e271828-) [inline readme.txt]: Is this actually what we want? The main plugin file has Version: 1.0. I believe it makes more sense to simply keep them aligned.

**Comment by phpwebdev11** [comment]
Closes #1 
Closes #3 
Closes #6 
Closes #8 

I have also completed the changes from the last commit.

**Pull request merged**